packages/frontend/src/components/Explorer/CustomDimensionModal/index.tsx

```diff
@@ -0,0 +1,204 @@
+            } else {
+                addCustomDimension({
+                    id: sanitizedId,
+                    name: values.customDimensionLabel,
+                    dimensionId: fieldId(item),
+                    binType: values.binType,
+                    binNumber: values.binConfig.fixedNumber.binNumber,
+                    table: item.table,
+
+                    // TODO: consider renaming some properties to match `addCustomMetric` logic
+                });
+
+                showToastSuccess({
+                    title: 'Custom dimension added successfully',
+                });
+            }
+        }
+
+        form.reset();
+        toggleModal();
+    });
+
+    return !!item ? (
+        <Modal
+            size="lg"
+            onClick={(e) => e.stopPropagation()}
+            opened={isOpen}
+            onClose={() => {
+                toggleModal(undefined);
+                form.reset();
+            }}
+            title={
+                <Title order={4}>
+                    {isEditing ? 'Edit' : 'Create'} Custom Dimension -{' '}
+                    {item.name}
+                </Title>
+            }
+        >
+            <form onSubmit={handleOnSubmit}>
+                <Stack>
+                    <TextInput
+                        label="Label"
+                        required
+                        placeholder="Enter custom dimension label"
+                        {...form.getInputProps('customDimensionLabel')}
+                    />
+
+                    <Radio.Group
+                        label="Bin type"
+                        withAsterisk
+                        required
+                        {...form.getInputProps('binType')}
+                    >
+                        <Group mt="md">
+                            <Radio
+                                value={BinType.FIXED_NUMBER}
+                                label="Fixed number of bins"
+                            />
+                            <Tooltip label="Coming soon">
+                                <Radio
+                                    disabled
+                                    value="fixed width"
+                                    label="Fixed Width"
+                                />
+                            </Tooltip>
+                            <Tooltip label="Coming soon">
+                                <Radio
+                                    disabled
+                                    value="custom range"
+                                    label="Custom Range"
+                                />
+                            </Tooltip>
+                        </Group>
+                    </Radio.Group>
+
+                    {form.values.binType === BinType.FIXED_NUMBER && (
+                        <NumberInput
+                            w={100}
+                            label="Bin number"
+                            required
+                            min={2}
```

Comment:
should we set a max ?

Reply:
![image](https://github.com/lightdash/lightdash/assets/1983672/69374d03-488d-4491-a354-49eb7ed0abc6)
 :see_no_evil: it doesn't error

Reply:
Let's leave it for now and then we can find through user feedback 😄  https://github.com/lightdash/lightdash/issues/7600